src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -251,8 +255,14 @@ public void setWebhookRegistered(boolean webhookRegistered) {
     protected void retrieve(@CheckForNull SCMSourceCriteria criteria, SCMHeadObserver observer,
                             @CheckForNull SCMHeadEvent<?> event,
                             TaskListener listener) throws IOException, InterruptedException {
+        handleRefreshingPRStore(event, listener);
+        gitSCMSource.accessibleRetrieve(criteria, observer, event, listener);
+    }
+
+    protected void handleRefreshingPRStore(@CheckForNull SCMHeadEvent<?> event,
+                                         TaskListener listener) {
         if (event == null) {
-            if (gitSCMSource.getTraits().stream().anyMatch(trait -> trait.getClass() == SelectBranchTrait.class)) {
+            if (getGitSCMSource().getTraits().stream().anyMatch(trait -> trait.getClass() == SelectBranchTrait.class)) {
```

Comment:
I like consistency, so I say either change to using `getGitSCMSource` everywhere, or not at all. I propose not at all, but your call.